src/plans/proximal_gradient_plan.jl

```diff
@@ -508,8 +508,11 @@ function (s::ProximalGradientMethodBacktrackingStepsize)(
 
         # Compute log_p(candidate_point) and its squared norm for the conditions
         log_p_q = inverse_retract(M, p, candidate_point, st.inverse_retraction_method)
-        squared_distance = distance(M, p, candidate_point)^2
-
+        squared_distance = if st.inverse_retraction_method == LogarithmicInverseRetraction()
```

Comment:
Wouldn't it make more sense to always calculate it as `norm(M, p, log_p_q)^2`? `norm` is generally faster and easier to compute than `distance`, and the result should be the same for `LogarithmicInverseRetraction`.